pkg/networkservice/chains/nsmgr/server.go

```diff
@@ -193,16 +200,18 @@ func NewServer(ctx context.Context, tokenGenerator token.GeneratorFunc, options
 			recvfd.NewServer(), // Receive any files passed
 			interpose.NewServer(&interposeRegistryServer),
 			filtermechanisms.NewServer(&urlsRegistryServer),
-			heal.NewServer(ctx,
-				heal.WithOnHeal(addressof.NetworkServiceClient(adapters.NewServerToClient(rv)))),
-			connect.NewServer(ctx,
-				client.NewClientFactory(
+			connect.NewServer(
+				client.NewClient(
+					ctx,
 					client.WithName(opts.name),
 					client.WithAdditionalFunctionality(
 						recvfd.NewClient(),
 						sendfd.NewClient(),
 					),
-				), opts.connectOptions...),
+					client.WithDialOptions(opts.dialOptions...),
+					client.WithDialTimeout(opts.dialTimeout),
+				),
```

Comment:
Shouldn't here be used `WithoutRefresh` option?

Reply:
You are correct :)

Reply:
Fixed